src/DataWrangling/metadata.jl

```diff
@@ -142,7 +142,7 @@ function Base.show(io::IO, metadata::Metadata)
 end
 
 # Treat Metadata as an array to allow iteration over the dates.
-Base.length(metadata::Metadata) = length(metadata.dates)
+Base.length(metadata::Metadata) = isnothing(metadata.dates) ? 1 : length(metadata.dates)
```

Comment:
```suggestion
Base.length(metadata::Metadata) = length(metadata.dates)
```
`Base.length` is defined below for `Metadatum` which should cover the case when `dates == nothing`